vault/vault.go

```diff
@@ -168,32 +167,24 @@ func (v Vault) getAccessToken() (string, error) {
 	}
 
 	request, err := json.Marshal(&rBody)
-
 	if err != nil {
-		log.Print("[WARN] marshalling grantRequest")
-		return "", err
+		return "", fmt.Errorf("marshalling token request body: %w", err)
 	}
 
 	url := v.urlFor("token", "")
 
 	response, err := handleResponse(http.Post(url, "application/json", bytes.NewReader(request)))
 	if err != nil {
-		log.Print("[DEBUG] grant response error:", err)
-		return "", err
+		return "", fmt.Errorf("fetching token: %w", err)
 	}
 
-	grant := struct {
-		AccessToken, TokenType string
-		ExpiresIn              int
-		// TODO cache the grant until it expires
-	}{}
-
-	if err = json.Unmarshal(response, &grant); err != nil {
-		log.Print("[INFO] parsing grant response:", err)
-		return "", err
+	// TODO: cache the token until it expires.
```

Comment:
@andrii-zakurenyi is this still applicable? If so we should create a github issue or other item to track this. mark this as closed after that or removing if appropriate.

Reply:
Yes, it still is. Created a PBI in the backlog